openshift/e2e-common.sh

```diff
@@ -119,6 +121,36 @@ function run_e2e_rekt_tests(){
   return $failed
 }
 
+function run_e2e_encryption_auth_tests(){
+  header "Running E2E Encryption and Auth Tests"
+
+  oc patch knativeeventing --type merge -n "${EVENTING_NAMESPACE}" knative-eventing --patch-file "${SCRIPT_DIR}/knative-eventing-encryption-auth.yaml"
+
+  images_file=$(dirname $(realpath "$0"))/images.yaml
+  make generate-release
+  cat "${images_file}"
+
+  oc wait --for=condition=Ready knativeeventing.operator.knative.dev knative-eventing -n "${EVENTING_NAMESPACE}" --timeout=900s
+
+  local regex="(.*TLS.*|.*OIDC.*)"
```

Comment:
Could we also run all tests (instead of only TLS & OIDC ones) to make sure everything works with TLS & OIDC enabled?

Reply:
I don't think that will work by design since for TLS tests you need to inject TLS specific options on the environment as well as for eventshub source

Reply:
IIUC then this would require to update the OIDC tests to configure e.g. the eventshub to use TLS?

Reply:
I think we would need to make OIDC _features_ "TLS aware" the part that might need tweaking is the part where we configure the destinations pointing to events with the usual:
```
d.CACerts = eventshub.GetCACerts(ctx)
```

in case the CA certs is configured it will be set otherwise they should continue to work on non TLS tests as it would be nil.

Then, we could have a rekt flag that autoconfigures eventshub to deploy TLS resources

Reply:
We only have to keep this in mind, when we move the OIDC tests from test/auth to test/rekt or test/experimental.
And can you give a few more details, what you mean with 

> Then, we could have a rekt flag that autoconfigures eventshub to deploy TLS resources

Reply:
Basically this https://github.com/openshift-knative/eventing/pull/400#discussion_r1371505133
> IIUC then this would require to update the OIDC tests to configure e.g. the eventshub to use TLS?

Reply:
> We only have to keep this in mind, when we move the OIDC tests from test/auth to test/rekt or test/experimental. And can you give a few more details, what you mean with

yeah, even though I would do this from the start 

> 
> > Then, we could have a rekt flag that autoconfigures eventshub to deploy TLS resources